common/common.go

```diff
@@ -151,7 +151,8 @@ type (
 
 	// reply to getblock verbose=1 (json includes txid list)
 	ZcashRpcReplyGetblock1 struct {
-		Tx []string
+		Hash string
+		Tx   []string
```

Comment:
Is this required to fix the race condition or is it an independent change? I would have thought it breaks compatibility for RPC clients?

Reply:
It's needed for the race condition fix. This is just an internal data structure for parsing zcashd's RPC reply. Previously the code was ignoring `hash` field, after this diff it will be parsed and available.

Reply:
Right, the way the Go JSON parsing package works, you can specify which object fields you want access to, and it will discard everything else. So all this change does is to provide us with the block hash. No changes to the wire format or the gRPC interface.